src/lib.rs

```diff
@@ -692,6 +746,54 @@ impl<R: ?Sized> Rng for Box<R> where R: Rng {
     }
 }
 
+/// Trait for casting types to byte slices
+pub trait AsByteSlice {
+    /// Return a mutable reference to self as a byte slice
+    fn as_byte_slice<'a>(&'a mut self) -> &'a mut [u8];
```

Comment:
I don't know for sure (still a novice and no computer right now), but is it possible to end up with two mutable references to the same memory after this function?

Reply:
The reference passed in will be locked until after the result is destroyed thanks to Rust's lifetime analysis. It's actually hard to get this wrong in Rust.